src/proto/uprotocol/uuri.rs

```diff
@@ -11,38 +11,617 @@
+    ///     })
+    ///     .into(),
+    ///     resource: Some(UResource {
+    ///         name: "rpc".to_string(),
+    ///         instance: Some("raise".to_string()),
+    ///         ..Default::default()
+    ///     })
+    ///     .into(),
+    ///     authority: None.into(),
+    ///     ..Default::default()
+    /// };
+    ///
+    /// let uri_from = UUri::try_from("/example.com//rpc.raise").unwrap();
+    /// assert_eq!(uri, uri_from);
+    /// ````
+    fn try_from(uri: &str) -> Result<Self, Self::Error> {
+        UUri::from_str(uri)
     }
 }
 
 impl TryFrom<UUri> for Vec<u8> {
     type Error = SerializationError;
 
     fn try_from(value: UUri) -> Result<Self, Self::Error> {
-        MicroUriSerializer::serialize(&value)
+        MicroUriSerializer::serialize(&value).map_err(|e| SerializationError::new(e.to_string()))
     }
 }
 
 impl TryFrom<Vec<u8>> for UUri {
     type Error = SerializationError;
 
     fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
-        MicroUriSerializer::deserialize(value)
+        MicroUriSerializer::deserialize(value).map_err(|e| SerializationError::new(e.to_string()))
+    }
+}
+
+impl UUri {
+    /// Builds a fully resolved `UUri` from the serialized long format and the serialized micro format.
+    ///
+    /// # Arguments
+    /// * `long_uri` - uri serialized as a string.
+    /// * `micro_uri` - uri serialized as a byte slice.
+    ///
+    /// # Returns
+    /// If successful, returns an UUri object serialized from the input formats. Returns `SerializationError` if the deserialization
+    /// fails or the resulting uri cannot be resolved.
+    pub fn build_resolved(long_uri: &str, micro_uri: &[u8]) -> Result<UUri, SerializationError> {
+        if long_uri.is_empty() && micro_uri.is_empty() {
+            return Err(SerializationError::new("Input uris are empty"));
+        }
+
+        let long_uri = UUri::from_str(long_uri)?;
+        let micro_uri = UUri::try_from(micro_uri.to_vec())?;
+
+        let mut auth = micro_uri.authority.unwrap_or_default();
+        let mut ue = micro_uri.entity.unwrap_or_default();
+        let mut ure = long_uri.resource.unwrap_or_default();
+
+        if let Some(authority) = long_uri.authority.as_ref() {
+            if let Some(name) = authority.get_name() {
+                auth.name = Some(name.to_owned());
+            }
+        }
+        if let Some(entity) = long_uri.entity.as_ref() {
+            ue.name = entity.name.clone();
+        }
+        if let Some(resource) = micro_uri.resource.as_ref() {
+            ure.id = resource.id;
+        }
+
+        let uri = UUri {
+            authority: Some(auth).into(),
+            entity: Some(ue).into(),
+            resource: Some(ure).into(),
+            ..Default::default()
+        };
+
+        if UriValidator::is_resolved(&uri) {
+            Ok(uri)
+        } else {
+            Err(SerializationError::new(format!(
+                "Could not resolve uri {uri}"
```

Comment:
If we do this, wouldn't we only print the long form via [impl TryFrom<&UUri> for String](https://github.com/eclipse-uprotocol/up-rust/pull/49/files#r1498198835)?

Would it be preferable to use the `Debug` output here instead `{:?}` so that we can read from the error all the contents of the UUri, including the numeric portions?

WDYT?

Reply:
Sure, let's put that it. If it gets on peoples nerves, we can always reduce verbosity again